### PR TITLE
[SPARK-6200] [SQL] Add a manager for dialects

### PR DESCRIPTION
**Pull request by the-sea**
This PR added an independent dialect manager.

support sql statement to manage dialect:

SHOW [EXTENDED] DIALECT -> show dialect name that is being used,if there is extended keyword,show the extended info.

SHOW [EXTENDED] DIALECTS -> show all dialects

CREATE DIALECT dialect_name USING dialect_full_classname -> add a new dialect with name "dialect_name" using the class that "dialect_full_classname" represented

DROP DIALECT dialect_name -> drop a dialect

USE DIALECT dialect_name -> switch dialect

### CODE REVIEW

**Comment by pwendell** [comment]
It might be simpler to just accept a dialect via a configuration option. Is the idea here that someone might switch dialects multiple times in a single session?

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 1]
```diff
@@ -20,6 +20,8 @@ package org.apache.spark.sql
 import java.beans.Introspector
 import java.util.Properties
 
+import org.apache.spark.sql.dialect.{DialectManager, DefaultDialectManager}
```
Comment (scwf) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: format issue

File: sql/core/src/main/scala/org/apache/spark/sql/execution/commands.scala [hunk 2]
```diff
@@ -208,6 +209,87 @@ case class DescribeCommand(
 }
 
 /**
+ * :: DeveloperApi ::
+ */
+@DeveloperApi
+case class ShowDialectsCommand(
+                                isExtended: Boolean,
+                                isCurrent: Boolean) extends RunnableCommand {
```
Comment (scwf) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/commands.scala]: should be Indentation

**Comment by scwf** [comment]
just some minor comment, this looks good to me

File: sql/core/src/main/scala/org/apache/spark/sql/execution/commands.scala (hunk 2 above)
Comment (haiyangsea) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/commands.scala]: updated,thank you for your review.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQlDialect.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+package org.apache.spark.sql.hive
+
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.dialect.Dialect
+
+/**
+ * Created by Sally on 2015/3/8.
+ */
```
Comment (scwf) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQlDialect.scala]: delete this

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQlDialect.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.dialect.Dialect
+
+object HiveQlDialect extends Dialect {
+  val name = "hiveql"
+
+  override def description = "Hive query language dialect"
+
+  override def parse(sql: String): LogicalPlan = {
+    HiveQl.parseSql(sql)
```
Comment (scwf) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQlDialect.scala]: we should first parse sql string by ddlparser

Comment (haiyangsea) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQlDialect.scala]: yeah,you are right.updated

**Pull request closed without merging**